### DISCUSSION TITLE
Chakra ui tabs when used with canvas is not rendering?

### DISCUSSION THREAD

**Discussion by rohitkrishna094** [chakra-ui/chakra-ui#3487] [Q&A] [answered]
I have also asked this question on [stackoverflow](https://stackoverflow.com/questions/66424473/react-chakra-ui-canvas-width-and-height-initializing-to-zero-and-unable-to-rende), adding here for reference

Here is the [codesandbox link](https://codesandbox.io/s/vigorous-glade-ohh5c?file=/src/App.js) 

and this is the same code:

```jsx
import "./styles.css";
import {
  Flex,
  Tab,
  TabList,
  TabPanel,
  TabPanels,
  Tabs
} from "@chakra-ui/react";
import { SketchField, Tools } from "react-sketch";

export default function App() {
  return (
    <Flex height="100vh" width="100vw">
      <Tabs width="100%" display="flex" flexDirection="column" defaultIndex={0}>
        <TabList>
          {["Play", "Draw"].map((val) => (
            <Tab key={Math.random()}>{val}</Tab>
          ))}
        </TabList>
        <TabPanels flex="1 1 auto">
          <TabPanel padding={0} height="100%">
            <h1>Hello world </h1>
          </TabPanel>
          <TabPanel padding={0} height="100%">
            <Flex height="100%" width="100%">
              <SketchField
                width="100%"
                height="100%"
                widthCorrection={0}
                tool={Tools.Pencil}
                lineColor="#3182CE"
                lineWidth={3}
              />
            </Flex>
          </TabPanel>
        </TabPanels>
      </Tabs>
    </Flex>
  );
}
```

The canvas component is not being shown because its height and width are set to zero for some reason. However if defaultIndex is 1(that is tab two is active), then it works. Not sure why.

I initially thought this was a problem with react-sketch, so I tried another canvas component(react-canvas-draw), the same problem persists with that component also.

**Comment by with-heart** [comment, marked as answer]
Not sure what the connection is to `canvas`, but your CodeSandbox works fine if you wrap the contents with `ChakraProvider` [as required](https://chakra-ui.com/docs/getting-started#setup-provider): https://codesandbox.io/s/vigorous-banach-9yedr?file=/src/App.js

**Reply by rohitkrishna094** [reply]
Thanks so much for the reply. Just now someone answered on stackoverflow asking me to use isLazy prop. Previously I didn't have it. Now its working. 

The solution was to use `isLazy` prop. We can mark the question as resolved.
